dns_server/src/crawler/mod.rs

```diff
@@ -0,0 +1,560 @@
+{
+    pub fn new(
+        config: CrawlerConfig,
+        conn: N::ConnectivityHandle,
+        sync: N::SyncingMessagingHandle,
+        storage: S,
+        command_tx: mpsc::UnboundedSender<ServerCommands>,
+    ) -> Result<Self, DnsServerError> {
+        // Load all persistent addresses
+        let tx = storage.transaction_ro()?;
+
+        let storage_version = tx.get_version()?;
+        if storage_version.is_some() && storage_version != Some(STORAGE_VERSION) {
+            return Err(DnsServerError::Other("Unexpected storage version"));
+        }
+
+        let mut addresses = BTreeMap::new();
+        for address in tx.get_addresses()?.iter().filter_map(|address| address.parse().ok()) {
+            Self::new_address(&mut addresses, address, true);
+        }
+        tx.close();
+
+        if storage_version.is_none() {
+            let mut tx = storage.transaction_rw()?;
+            tx.set_version(STORAGE_VERSION)?;
+            tx.commit()?;
+        }
+
+        // Add addresses that were specified from the command line as reachable
+        for address in config.add_node.iter() {
+            let address = address.parse()?;
+            Self::new_address(&mut addresses, address, true);
+        }
+
+        Ok(Self {
+            config,
+            conn,
+            sync,
+            addresses,
+            peers: BTreeMap::new(),
+            storage,
+            command_tx,
+        })
+    }
+
+    fn handle_conn_request(
+        &mut self,
+        _peer_id: N::PeerId,
+        request_id: N::PeerRequestId,
+        request: PeerManagerRequest,
+    ) {
+        match request {
+            PeerManagerRequest::AddrListRequest(_) => {
+                // Ignored
+            }
+            PeerManagerRequest::AnnounceAddrRequest(AnnounceAddrRequest { address }) => {
+                // TODO: Rate limit AnnounceAddrRequest requests
+                if let Some(address) = TransportAddress::from_peer_address(&address) {
+                    Self::new_address(&mut self.addresses, address, false);
+                }
+            }
+            PeerManagerRequest::PingRequest(PingRequest { nonce }) => {
+                let _ = self.conn.send_response(
+                    request_id,
+                    PeerManagerResponse::PingResponse(PingResponse { nonce }),
+                );
+            }
+        }
+    }
+
+    fn handle_conn_response(
+        &mut self,
+        _peer_id: N::PeerId,
+        _request_id: N::PeerRequestId,
+        _response: PeerManagerResponse,
+    ) {
+        // Ignore all
+    }
+
+    fn handle_outbound_accepted(
+        &mut self,
+        address: N::Address,
+        peer_info: PeerInfo<N::PeerId>,
+        _receiver_address: Option<PeerAddress>,
+    ) {
+        let address_data = self
+            .addresses
+            .get_mut(&address)
+            .expect("address in the Connecting state must be known");
+        let valid = peer_info.network == self.config.network;
```

Comment:
This doesn't seem to be a good strategy to do this check. The PeerInfo object should have something like `peer_info.is_compatible(&other_peer_info)` (or whatever else you find appropriate). There should not be any processing that's independent of p2p. It doesn't make sense that we're using the mintlayer p2p protocol, yet we invent our own rules here. Consider the effect of updates. 

I see here we're dealing with the crawler config. But being invasive like this is not the way to go. Please study this in a broader scope and let's see if we can come with a better compatibility check than just checking an arbitrary member variable in a struct.

Reply:
I added `is_compatible` to `PeerInfo`. For now it only checks `network`, but I added TODO to check version as well.